plugin/Makefile

```diff
@@ -84,24 +86,24 @@ info:
 
 clean:
 	@echo "=== Cleaning work files, images and plugin(s)..."
-	rm -rf ./rootfs
+	rm -rf $(TMP_LOC)
 	rm -f $(BINARY)
 	-docker plugin rm $(PLUGIN_NAME):$(PLUGIN_TAG) -f
+	-docker rm -vf $(TMP_CONTAINER)
 	-docker rmi $(TMP_IMAGE)
 
 
-plugin:
+plugin: clean
 	@echo "== building Docker image, unpacking to ./rootfs and creating plugin..."
 	cp $(BIN_LOC)/$(BINARY) .
 	docker build -q -t $(TMP_IMAGE) .
-	mkdir -p ./rootfs
 	docker create --name $(TMP_CONTAINER) $(TMP_IMAGE)
-	docker export  $(TMP_CONTAINER)  | tar -x -C ./rootfs
-	docker rm -vf  $(TMP_CONTAINER)
-	docker rmi $(TMP_IMAGE)
+	mkdir -p $(TMP_LOC)/rootfs
+	docker export  $(TMP_CONTAINER)  | tar -x -C $(TMP_LOC)/rootfs
+	cp config.json $(BINARY) $(TMP_LOC)
 	@echo "-- Creating  plugin $(PLUGIN_NAME):$(PLUGIN_TAG) ..."
-	docker plugin create $(PLUGIN_NAME):$(PLUGIN_TAG) .
-
+	docker plugin create $(PLUGIN_NAME):$(PLUGIN_TAG) $(TMP_LOC)
+	@rm -rf $(TMP_LOC)
```

Comment:
nit: L89 takes care this .. can it be redundant and removed? Please explain if I am missing something here.